app/streaming/input.cpp

```diff
@@ -884,30 +895,42 @@ void SdlInputHandler::rumble(unsigned short controllerNumber, unsigned short low
 
     // Stop the last effect we played
     if (m_GamepadState[controllerNumber].hapticEffectId >= 0) {
-        SDL_HapticDestroyEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId);
+        if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_LEFTRIGHT) {
+            SDL_HapticDestroyEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId);
+        } else if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_SIMPLERUMBLE) {
+            SDL_HapticRumbleStop(haptic);
+        }
     }
 
     // If this callback is telling us to stop both motors, don't bother queuing a new effect
     if (lowFreqMotor == 0 && highFreqMotor == 0) {
         return;
     }
 
-    SDL_HapticEffect effect;
-    SDL_memset(&effect, 0, sizeof(effect));
-    effect.type = SDL_HAPTIC_LEFTRIGHT;
+    if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_LEFTRIGHT) {
+        SDL_HapticEffect effect;
+        SDL_memset(&effect, 0, sizeof(effect));
+        effect.type = SDL_HAPTIC_LEFTRIGHT;
 
-    // The effect should last until we are instructed to stop or change it
-    effect.leftright.length = SDL_HAPTIC_INFINITY;
+        // The effect should last until we are instructed to stop or change it
+        effect.leftright.length = SDL_HAPTIC_INFINITY;
 
-    // SDL haptics range from 0-32767 but XInput uses 0-65535, so divide by 2 to correct for SDL's scaling
-    effect.leftright.large_magnitude = lowFreqMotor / 2;
-    effect.leftright.small_magnitude = highFreqMotor / 2;
+        // SDL haptics range from 0-32767 but XInput uses 0-65535, so divide by 2 to correct for SDL's scaling
+        effect.leftright.large_magnitude = lowFreqMotor / 2;
+        effect.leftright.small_magnitude = highFreqMotor / 2;
 
-    // Play the new effect
-    m_GamepadState[controllerNumber].hapticEffectId = SDL_HapticNewEffect(haptic, &effect);
-    if (m_GamepadState[controllerNumber].hapticEffectId >= 0) {
-        SDL_HapticRunEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId, 1);
+        // Play the new effect
+        m_GamepadState[controllerNumber].hapticEffectId = SDL_HapticNewEffect(haptic, &effect);
+        if (m_GamepadState[controllerNumber].hapticEffectId >= 0) {
+            SDL_HapticRunEffect(haptic, m_GamepadState[controllerNumber].hapticEffectId, 1);
+        }
+    } else if (m_GamepadState[controllerNumber].hapticMethod == GAMEPAD_HAPTIC_METHOD_SIMPLERUMBLE) {
+        SDL_HapticRumblePlay(haptic,
+                             std::min(1.0, std::max(highFreqMotor, lowFreqMotor)/65535.0),
+                             SDL_HAPTIC_INFINITY);
+        m_GamepadState[controllerNumber].hapticEffectId = 1;
```

Comment:
Setting `hapticEffectId` shouldn't be required for simple rumble, since we aren't using `SDL_HapticNewEffect()` so there's no allocated haptic effect to free later.

Reply:
Fixed